charts/alertmanager-snmp-notifier/values.yaml

```diff
@@ -0,0 +1,185 @@
+    - snmp-server:162
+
+  # SNMP authentication secrets, that may be instanciated by the chart, or may use an already created secret
+  # snmpCommunity: public
+  # snmpAuthenticationUsername: my_authention_username
+  # snmpAuthenticationPassword: my_authention_password
+  # snmpPrivatePassword: my_private_password
+  # snmpCommunitySecret:
+  #   name: existingsecret
+  #   key: community
+  # snmpAuthenticationUsernameSecret:
+  #   name: existingsecret
+  #   key: authenticationUsername
+  # snmpAuthenticationPasswordSecret:
+  #   name: existingsecret
+  #   key: authenticationPassword
+  # snmpPrivatePasswordSecret:
+  #   name: existingsecret
+  #   key: privatePassword
+
+  # snmpTemplates allows to customize the description of the traps, and add extra trap fields
+  # snmpTemplates:
+  #   description: |
+  #     {{- if .Alerts -}}
+  #     {{ len .Alerts }}/{{ len .DeclaredAlerts }} alerts are firing:
+
+  #     {{ range $severity, $alerts := (groupAlertsByLabel .Alerts "severity") -}}
+  #     Status: {{ $severity }}
+
+  #     {{- range $index, $alert := $alerts }}
+  #     - Alert: {{ $alert.Labels.alertname }}
+  #       Summary: {{ $alert.Annotations.summary }}
+  #       Description: {{ $alert.Annotations.description }}
+  #     {{ end }}
+  #     {{ end }}
+  #     {{ else -}}
+  #     Status: OK
+  #     {{- end -}}
+
+  #   extraFields:
+  #     - template: |
+  #         {{- if .Alerts -}}
+  #         Status: NOK
+  #         {{- else -}}
+  #         Status: OK
+  #         {{- end -}}
+  #     - template: |
+  #         This is a constant
+
+## Security context to be added to snmp-notifier pods
+securityContext:
+  {}
+  # fsGroup: 1000
+  # runAsUser: 1000
+  # runAsNonRoot: true
+
+## Security context to be added to snmp-notifier containers
+containerSecurityContext:
+  runAsNonRoot: true
+  runAsUser: 1000
+  readOnlyRootFilesystem: true
+
+## Additional labels to add to all resources
+customLabels:
+  {}
+  # app: snmp-notifier
+
+extraConfigmapMounts:
+  []
+  # - name: snmp-notifier-configmap
+  #   mountPath: /run/secrets/snmp-notifier
+  #   subPath: snmp.yaml # (optional)
+  #   configMap: snmp-notifier-configmap-configmap
+  #   readOnly: true
+  #   defaultMode: 420
+
+## Additional secret mounts
+# Defines additional mounts with secrets. Secrets must be manually created in the namespace.
+extraSecretMounts:
+  []
+  # - name: secret-files
+  #   mountPath: /run/secrets/snmp-notifier
+  #   secretName: snmp-notifier-secret-files
+  #   readOnly: true
+  #   defaultMode: 420
+
+## For RBAC support:
+rbac:
+  # Specifies whether RBAC resources should be created
+  create: true
+
+serviceAccount:
+  # Specifies whether a ServiceAccount should be created
+  create: true
+
+  # The name of the ServiceAccount to use.
+  # If not set and create is true, a name is generated using the fullname template
+  name:
+
+resources:
+  {}
+  # limits:
+  #   memory: 300Mi
+  # requests:
+  #   memory: 50Mi
+
+livenessProbe:
```

Comment:
liveness and readinessprobe should not be the same.
see: https://srcco.de/posts/kubernetes-liveness-probes-are-dangerous.html

Reply:
I am aware of this, but the app won't launch, if the SNMP server is unreachable or if there is a bad option, or bad template provided.
The HTTP server is the latest thing that is launched if everything is OK. That /health endpoint is therefore "the app is well configured and ready to receive alerts". I don't see what else could I check to differentiate the liveness from readiness.